File: 2022-2023/sourcecode/lab/lab10/ShoppyJava/app/src/main/java/pl/udu/uwr/pum/shoppyjava/fragments/add/AddFragment.java
package pl.udu.uwr.pum.shoppyjava.fragments.add;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;

import pl.udu.uwr.pum.shoppyjava.model.Item;
import pl.udu.uwr.pum.shoppyjava.viewmodel.ItemViewModel;
import pl.udu.uwr.pum.shoppyjava.databinding.FragmentAddBinding;

public class AddFragment extends Fragment {

    private FragmentAddBinding binding;

    private ItemViewModel viewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        viewModel = new ViewModelProvider(this).get(ItemViewModel.class);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentAddBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.saveButton.setOnClickListener(v -> {
            insertToDatabase(view);
        });
    }

    private void insertToDatabase(View view) {
        String name = binding.nameEditText.getText().toString();
        String quantity = binding.quantityEditText.getText().toString();

        if (TextUtils.isEmpty(name) && TextUtils.isEmpty(quantity)){
            binding.nameEditText.setError("Podaj nazwę");
            binding.quantityEditText.setError("Podaj ilość");
        } else {
            Item item = new Item(0, name, Integer.parseInt(quantity));
            viewModel.insert(item);
            Navigation.findNavController(view).navigate(AddFragmentDirections.actionAddFragmentToListFragment());
        }
    }
}

File: 2022-2023/sourcecode/lab/lab10/ShoppyKotlin/app/src/main/java/pl/udu/uwr/pum/shoppykotlin/fragments/add/AddFragment.kt
package pl.udu.uwr.pum.shoppykotlin.fragments.add

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import pl.udu.uwr.pum.shoppykotlin.data.Item
import pl.udu.uwr.pum.shoppykotlin.viewmodel.ItemViewModel
import pl.udu.uwr.pum.shoppykotlin.databinding.FragmentAddBinding

class AddFragment : Fragment() {

    private lateinit var binding: FragmentAddBinding

    private val itemViewModel: ItemViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentAddBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.saveButton.setOnClickListener {
            insertToDatabase()
        }
    }

    private fun insertToDatabase() {
        val name = binding.nameEditText.text.toString()
        val quantity = binding.quantityEditText.text.toString()

        if (name.isNotEmpty() && quantity.isNotEmpty()){
            val item = Item(0, name, quantity.toInt())
            itemViewModel.addItem(item)
            findNavController().navigate(AddFragmentDirections.actionAddFragmentToListFragment())
        } else{
            binding.nameEditText.error = "Podaj nazwę"
            binding.quantityEditText.error = "Podaj ilość"
        }
    }
}